libs/auth-api-lib/src/lib/services/delegations.service.ts

```diff
@@ -86,19 +90,119 @@ export class DelegationsService {
           toNationalId: d.toNationalId,
           fromNationalId: d.fromNationalId,
           fromName: d.fromDisplayName,
-          isFromCompany: d.isFromCompany,
           type: DelegationType.Custom,
           provider: DelegationProvider.Custom,
         },
     )
   }
+
+  async create(
+    nationalId: string,
+    delegation: DelegationDTO,
+  ): Promise<Delegation | null> {
+    this.logger.debug('Creating a new delegation')
+    const id = uuid()
+    await this.delegationModel.create({
+      id: id,
+      fromNationalId: nationalId,
+      ...delegation,
+    })
+    if (delegation.scopes) {
+      this.delegationScopeService.createMany(id, delegation.scopes)
+    }
+    return this.findOne(nationalId, id)
+  }
+
+  async update(
+    nationalId: string,
+    delegation: DelegationDTO,
+    id: string,
+  ): Promise<Delegation | null> {
+    this.logger.debug(`Updating a delegation with id ${id}`)
+    await this.delegationModel.update(
+      { ...delegation },
+      { where: { id: id, fromNationalId: nationalId } },
```

Comment:
If the id is found, but doesn't belong to the user ... a `404` should be returned. That logic should be implemented in the controller.

Reply:
Should we maybe throw an Exception like UnAuthorized rather than 404?

Reply:
We are throwing Unauthorized exception now

Reply:
Yeah both are valid,.. I usually go with 404 in these cases. The interpretation behind that is:
There is no delegation with that id that you can view... thus 404. In addition, I don't want the owner to know that there exists a record with that ID but it's owned by someone else. Therefore as far as the user (with nationalId) is concerned, a record with this ID doesn't exist.

But again, you can counter with arguments in favour of Unauthorized. So it's up to you how you interpret it, and then to try to enforce that rule to be consistent.